lib/connection.js

```diff
@@ -647,8 +662,9 @@ Connection.prototype.readBytes = function(buffer, length) {
 
 Connection.prototype.parseCString = function(buffer) {
   var start = this.offset;
-  while(buffer[this.offset++] !== 0) { }
-  return buffer.toString(this.encoding, start, this.offset - 1);
+  var end = indexOf(buffer, 0, start);
+  this.offset = end + 1;
```

Comment:
`this.offset` will be set to `0` if there wasn't a `'\0'` found, is that correct or should it maybe throw? :)